### DISCUSSION TITLE
EK60/EK80 NMEA datagram GPS extraction, NMEA sentences

### DISCUSSION THREAD

**Discussion by emiliom** [OSOceanAcoustics/echopype#231] [General]
Not sure this rises to the level of an issue yet, so I'm compiling some scattered discussions here.

- #210: GPS data from EK60/EK80 sometimes include different NMEA sentences. The spatial resolution appears to be different. Currently the default is to only save the GGA sentences unless others are specified by the user.
- #195:
  - (EM) In the current echopype code (latest release), the NMEA messages/datagrams used are 'GGA', 'GLL' and 'RMC', whereas in Don's branch based off the class-redesign branch, it's been changed to only use 'GGA'
  - (WJ) The NMEA message is definitely another point of discussion, which I mentioned in #210 . The short answer is that I found that the other messages contain low resolution data compared to GGA for the SH1707 EK60 dataset, and that these low-res messages are not in the other EK60 files I have, so I added that default but left the option to use others.

I think it'd be great to be able to extract and preserve the source-sentence code (GGA, etc) of the GPS data used, for reference and R&D. I haven't looked at the code specifically to see how easy this would be. For now I'm just recording that wish here.

**Comment by leewujung** [comment]
> I think it'd be great to be able to extract and preserve the source-sentence code (GGA, etc) of the GPS data used, for reference and R&D.

This is easy to do, basically to revert the changes I did to only select GGA. Thinking about it more though, we would need to add the message type, so that the lat/lon data can be easily selected in the downstream. It seems more convenient to add a data variable to store the message type, rather than separate out messages of different sources into different data variables (like `lat_GGA`, `lat_RMC` etc).

Thoughts?

**Comment by emiliom** [comment]
> This is easy to do

Great.

> It seems more convenient to add a data variable to store the message type, rather than separate out messages of different sources into different data variables (like lat_GGA, lat_RMC etc).

Yeah, that's what I was thinking (data variable storing the message type / code).

**Comment by emiliom** [comment]
Circling back to this topic, and just wanted to know where things stand. Has the default GPS datagram been changed? I have a vague memory of a discussion about this in the last several weeks, but nothing concrete.

Also, I spent a few minutes trying to navigate the code to figure out where this default is currently set, but was not successful. Could someone point me to a couple of relevant lines in the code? I assume that'd be in the class-redesign branch.

Thanks.

**Comment by emiliom** [comment]
I think I can now answer the first part of my question, about whether the default has changed. I didn't remember that there was an issue on this topic (#232); and from there I found PR #236 from Jan 19, which lists this change: "default NMEA to save all GGA GLL RMC sentences (which contain GPS data)".
